### PR TITLE
[Bug] When using view, make toSql method generates the final sql

### LINKED ISSUE #6735: [Bug] Cache cannot sense the modification of the view
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

rebase on branch-0.13

### What's Wrong?

Modifying the `VIEW` is equivalent to modifying the SQL, but because the SQL string has not changed, the query will hit the cached result before the `VIEW` is modified.

PR: [6687](https://github.com/apache/incubator-doris/pull/6687), by adding `updateTime` to `VIEW` and using `originSql + viewUpdateTime` as the cache key, this problem can be solved, but it is not very good.

### What You Expected?

> Note: There are two types of views, one is the view created by `create view ...`, which will be saved in the FE metadata. The other is the use of the `WITH` statement in SQL. After the SQL is parsed, a `localView` will be generated, which will not be saved in the metadata.

1. For `localView`, it does not need to be expanded into a statement, because `toSql` method already contains `WITH` statement. (However, there should be a bug in the with content display. When a `UNION` is encountered, the with statement is not displayed. The PR has been fixed)
2. For normal VIEW, it needs to be converted into a final statement
3. For the normal VIEW contained in `localView`, the normal VIEW will be expanded in the WITH statement

### How to Reproduce?

Omit

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zenoyang**
Fix #6735 

## Proposed changes

1. Fix the problem that the WITH statement cannot be printed when `UNION` is included in SQL
2. In the `toSql` method, convert the normal VIEW into the final statement
3. Replace `selectStmt.originSql` with `selectStmt.toSql`

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by morningman — COMMENTED** [review]
Please add UT for this case

**Comment by zenoyang** [comment]
> Please add UT for this case

Done. Please review again, thank you 😀

File: fe/fe-core/src/main/java/org/apache/doris/qe/cache/SqlCache.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public void setCacheInfo(CacheAnalyzer.CacheTable latestTable) {
 
     public InternalService.PFetchCacheResult getCacheData(Status status) {
         InternalService.PFetchCacheRequest request = InternalService.PFetchCacheRequest.newBuilder()
-                .setSqlKey(CacheProxy.getMd5(selectStmt.getOrigStmt().originStmt))
+                .setSqlKey(CacheProxy.getMd5(selectStmt.toSql()))
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/cache/SqlCache.java:44]: Why use `toSql()` can solve the issue #6735?

**Comment by xinghuayu007** [comment]
This solution is elegant！

**Review by morningman — APPROVED** [review]
lgtm

**Pull request merged**